### ISSUE TITLE
Could you share the training log for efficientdet d0

### ISSUE THREAD

**Issue by LinHungShi** [google/automl#380]
Hi, thanks for you hard work. I am implementing a pytorch version and having some problem with reproducing the performance. Could you please share the training log for tensorboard for efficientdet-d0? Thanks

**Comment by mingxingtan** [comment]
Here you go: https://storage.googleapis.com/cloud-tpu-checkpoints/efficientdet/coco/tb/efficientdet-d0-tb.tgz

**Comment by subhankar-ghosh** [comment]
This is helpful. Could you please share the training logs for efficientdet-d3? Thanks

**Comment by LinHungShi** [comment]
@mingxingtan  Thanks for sharing the log. After looking into the log, I am a little bit confused with the learning rate scheduling. It seems you're using batch size 128, which makes the adjusted learning 0.16.
Since the num_step_per_epoch is 120000(by default), the lr_warmup_step should be 937.5(120000/128). However I saw that the learning rate peeks at 1333 iterations. Could you let me know which part of my calculation is wrong?

**Comment by mingxingtan** [comment]
Hi @LinHungShi your calculation is correct, the peak LR is around 940:

![image](https://user-images.githubusercontent.com/6027221/81958712-3646ea80-95c3-11ea-9be4-92d31591fe49.png)

Notably, you need to unselect the smoothing to see the exact value:

![image](https://user-images.githubusercontent.com/6027221/81958855-6abaa680-95c3-11ea-8b4d-e76d2f4c15ba.png)

**Comment by LinHungShi** [comment]
@mingxingtan what I saw from the tensorboard looks like the following
![Screenshot from 2020-05-14 12-01-18](https://user-images.githubusercontent.com/12586217/81974921-1b807000-95db-11ea-9cb8-1e0f8f6f9aad.png)
![Screenshot from 2020-05-14 12-01-36](https://user-images.githubusercontent.com/12586217/81974931-20ddba80-95db-11ea-92f1-688812725b63.png)

**Comment by mingxingtan** [comment]
Since I couldn't reproduce the issue, closing the issue for now.

**Comment by LinHungShi** [comment]
Hi @mingxingtan I have a few questions regarding the training log.
1. I found that the time interval between any two consecutive points is very small. In total_loss, for instance, step 17.32k is reached at 11:27:01 while 17.54k is reached at 11:27:18. Consider that the batch size is 128, is it normal that training so fast? 
2. Also, since you're using distributed training, is the training log derived from single GPU/TPU or is it the result after the aggregation from different cores?
3.  I am trying to reproduce the model in other framework but I found that your training loss(including cls loss and reg loss) fluctuates a lot while eval goes up smoothly. This is opposite to my training log where training loss decreases smoothly but eval moves zigzagly. I know it'd be hard to debug without seeing the code, but do you have any idea where might be wrong in my training?

**Comment by mingxingtan** [comment]
1. I used 32 TPU cores, so each core only needs to process 4 examples in each step.
2. It is sync training and results are aggregated.
3. yeah, it is hard to tell without seeing the code or log.

**Comment by XhqGlorry11** [comment]
@LinHungShi Hello, have u succeeded in reproducing training on COCO in pytorch. Since the link shared by author has expired, can u share the training log? Thanks!

**Comment by mingxingtan** [comment]
Hmm, not sure why the previous links expire. FYI, here are the two tensorboard.dev links for D0 and D4:

D0: https://tensorboard.dev/experiment/4IzZoQyAQLSManQfwaHyyQ/#scalars  (peak val mAP: 34.37)
D4: https://tensorboard.dev/experiment/KkM6IeARTmmkJPiB9gJoaw/#scalars   (peak val mAP: 49.32)

I believe tensorboard.dev links won't expire.

**Comment by XhqGlorry11** [comment]
@mingxingtan Thanks for sharing. According to your training log, cls&reg loss starts to decrease immediately as the training begins. But in my training, cls loss starts around 1.4 and oscillates, while cls loss 0.116. Is that normal and I should continue training or there is something wrong?

**Comment by XhqGlorry11** [comment]
I am using resnet50 as backbone and load the pretrained weights, btw.
